src/main/java/net/openhft/chronicle/threads/BusyTimedPauser.java

```diff
@@ -47,27 +52,59 @@ public void pause() {
         Jvm.nanoPause();
     }
 
+    /**
+     * Attempts to pause the thread with a specified timeout. If the pause exceeds the specified duration,
+     * a {@link TimeoutException} is thrown, indicating the timeout has elapsed without resumption of operations.
+     *
+     * @param timeout  the maximum time to wait before throwing an exception
+     * @param timeUnit the unit of time for the timeout parameter
+     * @throws TimeoutException if the wait exceeds the specified timeout duration
+     */
     @Override
     public void pause(long timeout, TimeUnit timeUnit) throws TimeoutException {
         if (time == Long.MAX_VALUE)
             time = System.nanoTime();
-        if (time + timeUnit.toNanos(timeout) - System.nanoTime() < 0)
-            throw new TimeoutException();
+        if (System.nanoTime() - time > timeUnit.toNanos(timeout))
```

Comment:
Would be good to highlight this improvement in the linked issue - also for this info to show up in release notes the commit will need to contain `fixes #251`